### PR TITLE
Remove need for external H3 crate/type

### LINKED ISSUE #18: Use raw u64 for index/tree-key?
There's another H3 crate now, and prescribing a particular crate's  H3 cell/index type seems overly restrictive.

CC @madninja

### PR DESCRIPTION
**Pull request by JayKickliter**
This PR removes the dependency on an external H3 crate by implementing its own [`Index`] type. 

Remaining work:

- [x] add proper error type for u64 -> Cell conversion. 
    - This PR no longer allows the user to insert invalid cells into a tree
- [x] bikeshed API naming
- [x] I already realize I should have called the type Cell
- [x] cleanup doc tests which still rely on h3ron for geo->h3 conversions.
    - [x] remove all unwraps from documentation code 
- [x] rename index mod to cell and both module and type-level docs
- [x] the Cell bitfield exposes a weird API, and it really should be Index. Name it back to Index and wrap it in a `struct Cell(Index)`.

Closes #18 

[`Index`]: https://github.com/JayKickliter/HexTree/blob/jsk/use-agnostic-index-type/src/index.rs
[digits iterator]: https://github.com/JayKickliter/HexTree/blob/jsk/use-agnostic-index-type/src/digits.rs

### CODE REVIEW

**Comment by JayKickliter** [comment]
@madninja I don't have a whole lot of time to clean this up to a proper release ATM, but it's basically there. I can reprioritize if needed though.

**Comment by madninja** [comment]
> @madninja I don't have a whole lot of time to clean this up to a proper release ATM, but it's basically there. I can reprioritize if needed though.

Nope. Fine for now. We can get back  to this

File: src/index.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    u8; pub res10digit, set_res10digit : 17, 15;
+    u8; pub res11digit, set_res11digit : 14, 12;
+    u8; pub res12digit, set_res12digit : 11,  9;
+    u8; pub res13digit, set_res13digit :  8,  6;
+    u8; pub res14digit, set_res14digit :  5,  3;
+    u8; pub res15digit, set_res15digit :  2,  0;
+}
+
+impl Index {
+    pub fn from_raw(raw: u64) -> Result<Self, ()> {
```
Comment (JayKickliter) [inline src/index.rs:36]: Does a result make sense here? I think it does. Either the raw 64 is a valid H3 index or not. Either way, a proper Error impl type is needed.

File: src/index.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+bitfield::bitfield! {
+    /// An H3 index.
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    #[cfg_attr(
+        feature = "serde-support",
+        derive(serde::Serialize, serde::Deserialize),
+        serde(transparent)
+    )]
+    pub struct Index(u64);
```
Comment (JayKickliter) [inline src/index.rs:12]: Rename to `Cell` to better describe the fact that we constrain valid representations to only H3 cell mode.

File: src/index.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    u8; pub res10digit, set_res10digit : 17, 15;
+    u8; pub res11digit, set_res11digit : 14, 12;
+    u8; pub res12digit, set_res12digit : 11,  9;
+    u8; pub res13digit, set_res13digit :  8,  6;
+    u8; pub res14digit, set_res14digit :  5,  3;
+    u8; pub res15digit, set_res15digit :  2,  0;
+}
+
+impl Cell {
+    pub fn from_raw(raw: u64) -> Result<Self, ()> {
```
Comment (JayKickliter) [inline src/index.rs:36]: Does a result make sense here? I think it does. Either the raw 64 is a valid H3 index or not. Either way, a proper Error impl type is needed.

File: src/index.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        idx.base_cell() < 122
+        {
+            Ok(idx)
+        } else {
+            Err(())
+        }
+    }
+
+    pub fn parent(&self, res: u8) -> Option<Self> {
+        if self.resolution() < res {
```
Comment (JayKickliter) [inline src/index.rs:53]: I don't fully agree with clippy here, but I'll make the change

**Review by madninja — APPROVED** [review]

**Pull request merged**